eessi/reframe/eessi_checks/applications/gromacs_check.py

```diff
@@ -21,16 +22,17 @@ class GROMACS_EESSI(gromacs_check):
     @run_after('init')
     def run_after_init(self):
         """hooks to run after the init phase"""
-        hooks.filter_tests_by_device_type(self, required_device_type=self.nb_impl)
+        hooks.filter_valid_systems_by_device_type(self, required_device_type=self.nb_impl)
         hooks.set_modules(self)
         hooks.set_tag_scale(self)
 
     @run_after('init')
     def set_tag_ci(self):
+        """Set tag CI on first benchmark"""
         if self.benchmark_info[0] == 'HECBioSim/hEGFRDimer':
-            self.tags.add('CI')
+            self.tags.add(TAGS['CI'])
 
-    @run_after('init')
+    @run_after('setup')
```

Comment:
moving this hook to after the setup phase avoids running it on all tests instead of only on the ones that will actually run.